Dockerfile

```diff
@@ -1,33 +1,27 @@
-FROM golang:1.13.5 AS base-env
+FROM alpine:3.13
 
-WORKDIR /src/
-
-COPY go.mod .
-COPY go.sum .
-RUN go mod download
-
-COPY . .
-
-FROM base-env AS build-env
-ENV BUILD_TARGET=/src/nri-kube-events
-ENV CGO_ENABLED=0
-RUN make compile
-
-FROM alpine:3.13 AS final
+# Set by docker automatically
+# If building with `docker build`, make sure to set GOOS/GOARCH explicitly when calling make:
+# `make compile GOOS=something GOARCH=something`
+# Otherwise the makefile will not append them to the binary name and docker build wil fail.
+ARG TARGETOS
+ARG TARGETARCH
 
 WORKDIR /app
 
 RUN apk add --no-cache --upgrade \
-    ca-certificates \
+    tini ca-certificates \
     && addgroup -g 2000 nri-kube-events \
     && adduser -D -H -u 1000 -G nri-kube-events nri-kube-events
-USER nri-kube-events
-
-COPY --from=build-env /src/nri-kube-events .
 EXPOSE 8080
 
+ADD --chmod=755 bin/nri-kube-events-${TARGETOS}-${TARGETARCH} ./
+RUN mv nri-kube-events-${TARGETOS}-${TARGETARCH} nri-kube-events
```

Comment:
just out of curiosity, couldn't the binary be renamed in the ADD line above?

Reply:
Because docker is ~~stupid~~ limited and that doesn't work :(

In my local machine, doing
`ADD --chmod=755 bin/nri-kube-events-${TARGETOS}-${TARGETARCH} /app/nri-kube-events` works fine. However, on the CI this will result in the binary ending up in `/app/nri-kube-events/kube-events-linux-amd64`. Reason for this is likely that this was changed/fixed in some docker version that the GHA ubuntu env doesn't have. This happened with both `COPY` and `ADD`, so I went for the quick hack.